lib/internal/assert/assertion_error.js

```diff
@@ -41,281 +35,185 @@ const kReadableOperator = {
-    } else {
-      other = a;
-    }
-    ArrayPrototypePop(actualLines);
-    ArrayPrototypePop(expectedLines);
-    if (actualLines.length === 0 || expectedLines.length === 0)
-      break;
-    a = actualLines[actualLines.length - 1];
-    b = expectedLines[expectedLines.length - 1];
+  return true;
+}
+
+function getColoredMyersDiff(actual, expected) {
+  const header = `${colors.green}actual${colors.white} ${colors.red}expected${colors.white}`;
+  const skipped = false;
+
+  const diff = myersDiff(StringPrototypeSplit(actual, ''), StringPrototypeSplit(expected, ''));
+  let message = printSimpleMyersDiff(diff);
+
+  if (skipped) {
+    message += '...';
   }
 
-  const maxLines = MathMax(actualLines.length, expectedLines.length);
-  // Strict equal with identical objects that are not identical by reference.
-  // E.g., assert.deepStrictEqual({ a: Symbol() }, { a: Symbol() })
-  if (maxLines === 0) {
-    // We have to get the result again. The lines were all removed before.
-    const actualLines = StringPrototypeSplit(actualInspected, '\n');
-
-    // Only remove lines in case it makes sense to collapse those.
-    // TODO: Accept env to always show the full error.
-    if (actualLines.length > 50) {
-      actualLines[46] = `${colors.blue}...${colors.white}`;
-      while (actualLines.length > 47) {
-        ArrayPrototypePop(actualLines);
+  return { message, header, skipped };
+}
+
+function getStackedDiff(actual, expected) {
+  const isStringComparison = typeof actual === 'string' && typeof expected === 'string';
+
+  let message = `\n${colors.green}+${colors.white} ${actual}\n${colors.red}- ${colors.white}${expected}`;
+  const stringsLen = actual.length + expected.length;
+  const maxTerminalLength = process.stderr.isTTY ? process.stderr.columns : 80;
+  const showIndicator = isStringComparison && (stringsLen <= maxTerminalLength);
+
+  if (showIndicator) {
+    let indicatorIdx = -1;
+
+    for (let i = 0; i < actual.length; i++) {
+      if (actual[i] !== expected[i]) {
+        // Skip the indicator for the first 2 characters because the diff is immediately apparent
+        // It is 3 instead of 2 to account for the quotes
+        if (i >= 3) {
+          indicatorIdx = i;
+          break;
+        }
+        break;
       }
     }
 
-    return `${kReadableOperator.notIdentical}\n\n` +
-           `${ArrayPrototypeJoin(actualLines, '\n')}\n`;
+    if (indicatorIdx !== -1) {
+      message += `\n${StringPrototypeRepeat(' ', indicatorIdx + 2)}^`;
+    }
   }
 
-  // There were at least five identical lines at the end. Mark a couple of
-  // skipped.
-  if (i >= 5) {
-    end = `\n${colors.blue}...${colors.white}${end}`;
-    skipped = true;
-  }
-  if (other !== '') {
-    end = `\n  ${other}${end}`;
-    other = '';
+  return { message };
+}
+
+function getSimpleDiff(originalActual, actual, originalExpected, expected) {
+  const stringsLen = actual.length + expected.length;
+  if (stringsLen <= kMaxShortStringLength) {
```

Comment:
We should keep the special handling for zero as before:

(actual !== 0 || expected !== 0)) { // -0 === +0

The output would otherwise be confusing because it would effectively be a different comparison.